files/grest/rpc/pool/pool_info.sql

```diff
@@ -1,118 +1,108 @@
+    block_data.op_cert_counter,
+    active_stake.as_sum,
+    block_data.cnt,
+    live.stake,
+    live.delegators,
+    ROUND((live.stake / _total_supply) * 100, 2)
+  FROM
+    grest.pool_info_cache AS pic
+  LEFT JOIN
+    public.pool_offline_data AS pod ON pod.pmr_id = pic.meta_id
+  LEFT JOIN LATERAL(
     SELECT
-        pic.pool_id_bech32,
-        pic.pool_id_hex,
-        pic.active_epoch_no,
-        pic.vrf_key_hash,
-        pic.margin,
-        pic.fixed_cost,
-        pic.pledge,
-        pic.reward_addr,
-        pic.owners,
-        pic.relays,
-        pic.meta_url,
-        pic.meta_hash,
-        pod.json,
-        pic.pool_status,
-        pic.retiring_epoch,
-        encode(opcert.op_cert::bytea, 'hex'),
-        opcert.op_cert_counter,
-        active_stake.as_sum,
-        blocks.cnt,
-        live.stake,
-        live.delegators,
-        ROUND((live.stake / _total_supply) * 100, 2)
-    FROM
-        grest.pool_info_cache AS pic
-    LEFT JOIN
-        public.pool_offline_data AS pod ON pod.pmr_id = pic.meta_id
-    LEFT JOIN LATERAL(
-        SELECT 
-            b.op_cert,
-            b.op_cert_counter
-        FROM 
-            public.block AS b
-        INNER JOIN 
-            public.slot_leader AS sl ON b.slot_leader_id = sl.id
-        WHERE
-            sl.pool_hash_id = pic.pool_hash_id
-        ORDER BY 
-            b.slot_no DESC
-        LIMIT 1
-    ) opcert ON TRUE
-    LEFT JOIN LATERAL(
-        SELECT
-            SUM (es.amount)::lovelace AS as_sum
-        FROM 
-            public.epoch_stake AS es
-        WHERE 
-            es.epoch_no = _epoch_no
-            AND 
-            es.pool_id = pic.pool_hash_id
-    ) active_stake ON TRUE
-    LEFT JOIN LATERAL(
-        SELECT
-            COUNT (b.id) AS cnt
-        FROM 
-            public.block AS b
-        INNER JOIN 
-            public.slot_leader AS sl ON b.slot_leader_id = sl.id
-        WHERE
-            sl.pool_hash_id = pic.pool_hash_id
-            AND
-            b.epoch_no = _epoch_no
-    ) blocks ON TRUE
-    LEFT JOIN LATERAL(
-        SELECT
-            SUM (total_balance)::lovelace AS stake,
-            COUNT (stake_address) AS delegators
-        FROM
-            grest.stake_distribution_cache AS sdc
-        WHERE
-            sdc.pool_id = pic.pool_id_bech32
-    ) live ON TRUE
+      SUM(COUNT(b.id)) OVER () AS cnt,
+      b.op_cert,
+      b.op_cert_counter
+    FROM 
+      public.block AS b
+    INNER JOIN 
+      public.slot_leader AS sl ON b.slot_leader_id = sl.id
     WHERE
-        pic.pool_id_bech32 = _pool_bech32
+      sl.pool_hash_id = pic.pool_hash_id
+    GROUP BY
+	    b.op_cert,
+	    b.op_cert_counter
     ORDER BY
-        pic.tx_id DESC
-    LIMIT 1;
+	    b.op_cert_counter DESC
+    LIMIT 1
+  ) block_data ON TRUE
+  LEFT JOIN LATERAL(
+    SELECT
+      amount::lovelace AS as_sum
+    FROM 
+      grest.pool_active_stake_cache AS easc
+    WHERE 
+      easc.pool_id = pic.pool_id_bech32
```

Comment:
```suggestion
      easc.pool_id = pic.pool_id_bech32 AND
      easc.epoch_no = _epoch_no
```

Reply:
The cache table has an entry for every epoch, here we only want to look at the current epoch.